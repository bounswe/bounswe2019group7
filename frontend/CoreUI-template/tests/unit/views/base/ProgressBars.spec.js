import Vue from 'vue'
import { mount, shallowMount } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue'
import ProgressBars from '@/views/base/ProgressBars'

Vue.use(BootstrapVue)

jest.useFakeTimers()

describe('ProgressBars.vue', () => {
  it('has a name', () => {
    expect(ProgressBars.name).toMatch('progress-bars')
  })
  it('has a created hook', () => {
    expect(typeof ProgressBars.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof ProgressBars.data).toMatch('function')
    const defaultData = ProgressBars.data()
    expect(defaultData.counter).toBe(45)
  })
  it('is Vue instance', () => {
    const wrapper = mount(ProgressBars)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is ProgressBars', () => {
    const wrapper = mount(ProgressBars)
    expect(wrapper.is(ProgressBars)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(ProgressBars)
    expect(wrapper.find('header.card-header > div > strong').text()).toMatch('Bootstrap Progress')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(ProgressBars)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should be destroyed', () => {
    const wrapper = mount(ProgressBars)
    wrapper.destroy()
  })
  it('should have methods', () => {
    expect(typeof ProgressBars.methods.clicked  ).toEqual('function')
    expect(ProgressBars.methods.clicked()).toBeUndefined()
  })
})
