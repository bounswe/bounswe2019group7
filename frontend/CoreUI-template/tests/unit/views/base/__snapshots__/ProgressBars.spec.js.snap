// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBars.vue renders correctly 1`] = `
<div
  class="wrapper"
>
  <div
    class="animated fadeIn"
  >
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Bootstrap Progress
        </strong>
         
        <div
          class="card-header-actions"
        >
          <a
            class="card-header-action"
            href="https://bootstrap-vue.js.org/docs/components/progress"
            rel="noreferrer noopener"
            target="_blank"
          >
            <small
              class="text-muted"
            >
              docs
            </small>
          </a>
        </div>
      </div>
       
      <div>
        <b-progress-stub
          animated="true"
          max="100"
          precision="0"
          showprogress="true"
          value="45"
        />
         
        <b-progress-stub
          class="mt-1"
          max="100"
          precision="0"
          showvalue="true"
          value="0"
        >
          <b-progress-bar-stub
            value="27"
            variant="success"
          />
           
          <b-progress-bar-stub
            value="11.25"
            variant="warning"
          />
           
          <b-progress-bar-stub
            value="6.75"
            variant="danger"
          />
        </b-progress-stub>
         
        <b-btn-stub
          class="mt-4"
          event="click"
          routertag="a"
          tag="button"
          target="_self"
          type="button"
          variant="secondary"
        >
          Click me
        </b-btn-stub>
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
         
        <strong>
           Progress 
        </strong>
        <small>
          labels
        </small>
      </div>
       
      <div>
        <h6>
          No label
        </h6>
         
        <b-progress-stub
          class="mb-3"
          max="50"
          precision="0"
          value="33.333333333"
        />
         
        <h6>
          Value label
        </h6>
         
        <b-progress-stub
          class="mb-3"
          max="50"
          precision="0"
          showvalue="true"
          value="33.333333333"
        />
         
        <h6>
          Progress label
        </h6>
         
        <b-progress-stub
          class="mb-3"
          max="50"
          precision="0"
          showprogress="true"
          value="33.333333333"
        />
         
        <h6>
          Value label with precision
        </h6>
         
        <b-progress-stub
          class="mb-3"
          max="50"
          precision="2"
          showvalue="true"
          value="33.333333333"
        />
         
        <h6>
          Progress label with precision
        </h6>
         
        <b-progress-stub
          class="mb-3"
          max="50"
          precision="2"
          showprogress="true"
          value="33.333333333"
        />
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Progress 
        </strong>
        <small>
          width
        </small>
      </div>
       
      <div>
        <h6>
          Default width
        </h6>
         
        <b-progress-stub
          class="mb-3"
          max="100"
          precision="0"
          value="75"
        />
         
        <h6>
          Custom widths
        </h6>
         
        <b-progress-stub
          class="w-75 mb-2"
          max="100"
          precision="0"
          value="75"
        />
         
        <b-progress-stub
          class="w-50 mb-2"
          max="100"
          precision="0"
          value="75"
        />
         
        <b-progress-stub
          class="w-25"
          max="100"
          precision="0"
          value="75"
        />
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Progress 
        </strong>
        <small>
          height
        </small>
      </div>
       
      <div>
        <h6>
          Default height
        </h6>
         
        <b-progress-stub
          class="mb-3"
          max="100"
          precision="0"
          showprogress="true"
          value="75"
        />
         
        <h6>
          Custom heights
        </h6>
         
        <b-progress-stub
          class="mb-2"
          height="2rem"
          max="100"
          precision="0"
          showprogress="true"
          value="75"
        />
         
        <b-progress-stub
          class="mb-2"
          height="20px"
          max="100"
          precision="0"
          showprogress="true"
          value="75"
        />
         
        <b-progress-stub
          height="2px"
          max="100"
          precision="0"
          value="75"
        />
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Progress 
        </strong>
        <small>
          variants
        </small>
      </div>
       
      <div>
        <div
          class="row mb-1"
        >
          <div
            class="col-sm-2"
          >
            success:
          </div>
           
          <div
            class="col-sm-10 pt-1"
          >
            <b-progress-stub
              max="100"
              precision="0"
              value="75"
              variant="success"
            />
          </div>
        </div>
        <div
          class="row mb-1"
        >
          <div
            class="col-sm-2"
          >
            info:
          </div>
           
          <div
            class="col-sm-10 pt-1"
          >
            <b-progress-stub
              max="100"
              precision="0"
              value="75"
              variant="info"
            />
          </div>
        </div>
        <div
          class="row mb-1"
        >
          <div
            class="col-sm-2"
          >
            warning:
          </div>
           
          <div
            class="col-sm-10 pt-1"
          >
            <b-progress-stub
              max="100"
              precision="0"
              value="75"
              variant="warning"
            />
          </div>
        </div>
        <div
          class="row mb-1"
        >
          <div
            class="col-sm-2"
          >
            danger:
          </div>
           
          <div
            class="col-sm-10 pt-1"
          >
            <b-progress-stub
              max="100"
              precision="0"
              value="75"
              variant="danger"
            />
          </div>
        </div>
        <div
          class="row mb-1"
        >
          <div
            class="col-sm-2"
          >
            primary:
          </div>
           
          <div
            class="col-sm-10 pt-1"
          >
            <b-progress-stub
              max="100"
              precision="0"
              value="75"
              variant="primary"
            />
          </div>
        </div>
        <div
          class="row mb-1"
        >
          <div
            class="col-sm-2"
          >
            secondary:
          </div>
           
          <div
            class="col-sm-10 pt-1"
          >
            <b-progress-stub
              max="100"
              precision="0"
              value="75"
              variant="secondary"
            />
          </div>
        </div>
        <div
          class="row mb-1"
        >
          <div
            class="col-sm-2"
          >
            dark:
          </div>
           
          <div
            class="col-sm-10 pt-1"
          >
            <b-progress-stub
              max="100"
              precision="0"
              value="75"
              variant="dark"
            />
          </div>
        </div>
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Progress 
        </strong>
        <small>
          striped
        </small>
      </div>
       
      <div>
        <b-progress-stub
          class="mb-2"
          max="100"
          precision="0"
          striped="true"
          value="25"
          variant="success"
        />
         
        <b-progress-stub
          class="mb-2"
          max="100"
          precision="0"
          striped="true"
          value="50"
          variant="info"
        />
         
        <b-progress-stub
          class="mb-2"
          max="100"
          precision="0"
          striped="true"
          value="75"
          variant="warning"
        />
         
        <b-progress-stub
          class="mb-2"
          max="100"
          precision="0"
          striped="true"
          value="100"
          variant="danger"
        />
         
        <b-button-stub
          event="click"
          routertag="a"
          tag="button"
          target="_self"
          type="button"
          variant="secondary"
        >
          
          Remove Striped
        
        </b-button-stub>
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Progress 
        </strong>
        <small>
          animated
        </small>
      </div>
       
      <div>
        <b-progress-stub
          animated="true"
          class="mb-2"
          max="100"
          precision="0"
          striped="true"
          value="25"
          variant="success"
        />
         
        <b-progress-stub
          animated="true"
          class="mb-2"
          max="100"
          precision="0"
          striped="true"
          value="50"
          variant="info"
        />
         
        <b-progress-stub
          animated="true"
          class="mb-2"
          max="100"
          precision="0"
          striped="true"
          value="75"
          variant="warning"
        />
         
        <b-progress-stub
          animated="true"
          class="mb-3"
          max="100"
          precision="0"
          value="100"
          variant="danger"
        />
         
        <b-button-stub
          event="click"
          routertag="a"
          tag="button"
          target="_self"
          type="button"
          variant="secondary"
        >
          
          Stop Animation
        
        </b-button-stub>
      </div>
    </b-card-stub>
     
    <b-card-stub
      bodytag="div"
      footertag="footer"
      headertag="header"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        <i
          class="fa fa-align-justify"
        />
        <strong>
           Progress 
        </strong>
        <small>
          multiple bars
        </small>
      </div>
       
      <div>
        <b-progress-stub
          class="mb-3"
          max="100"
          precision="0"
          value="0"
        >
          <b-progress-bar-stub
            value="15"
            variant="primary"
          />
           
          <b-progress-bar-stub
            value="30"
            variant="success"
          />
           
          <b-progress-bar-stub
            value="20"
            variant="info"
          />
        </b-progress-stub>
         
        <b-progress-stub
          class="mb-3"
          max="100"
          precision="0"
          showprogress="true"
          value="0"
        >
          <b-progress-bar-stub
            value="15"
            variant="primary"
          />
           
          <b-progress-bar-stub
            value="30"
            variant="success"
          />
           
          <b-progress-bar-stub
            value="20"
            variant="info"
          />
        </b-progress-stub>
         
        <b-progress-stub
          class="mb-3"
          max="100"
          precision="0"
          showvalue="true"
          striped="true"
          value="0"
        >
          <b-progress-bar-stub
            value="15"
            variant="primary"
          />
           
          <b-progress-bar-stub
            value="30"
            variant="success"
          />
           
          <b-progress-bar-stub
            value="20"
            variant="info"
          />
        </b-progress-stub>
         
        <b-progress-stub
          class="mb-3"
          max="100"
          precision="0"
          value="0"
        >
          <b-progress-bar-stub
            showprogress="true"
            value="15"
            variant="primary"
          />
           
          <b-progress-bar-stub
            animated="true"
            showprogress="true"
            value="30"
            variant="success"
          />
           
          <b-progress-bar-stub
            showprogress="true"
            striped="true"
            value="20"
            variant="info"
          />
        </b-progress-stub>
      </div>
    </b-card-stub>
  </div>
</div>
`;
